src/Simplex/Messaging/Agent/Store/SQLite/Util.hs

```diff
@@ -1,382 +1,403 @@
-        INSERT INTO servers (host, port, key_hash) VALUES (?, ?, ?)
-        ON CONFLICT (host, port) DO UPDATE SET
-          host=excluded.host,
-          port=excluded.port,
-          key_hash=excluded.key_hash;
-      |]
-      srv
-    DB.queryNamed
-      conn
-      "SELECT server_id FROM servers WHERE host = :host AND port = :port"
-      [":host" := host, ":port" := port]
-  case r of
-    [Only serverId] -> return serverId
-    _ -> throwError SEInternal
+{- ORMOLU_DISABLE -}
+-- SQLite.Simple only has these up to 10 fields, which is insufficient for some of our queries
+instance (FromField a, FromField b, FromField c, FromField d, FromField e,
+          FromField f, FromField g, FromField h, FromField i, FromField j,
+          FromField k) =>
+  FromRow (a,b,c,d,e,f,g,h,i,j,k) where
+  fromRow = (,,,,,,,,,,) <$> field <*> field <*> field <*> field <*> field
+                         <*> field <*> field <*> field <*> field <*> field
+                         <*> field
+{- ORMOLU_ENABLE -}
+
+createRcvQueueAndConn :: DB.Connection -> ReceiveQueue -> IO ()
+createRcvQueueAndConn dbConn rcvQueue =
+  DB.withTransaction dbConn $ do
+    upsertServer_ dbConn (server (rcvQueue :: ReceiveQueue))
+    insertRcvQueue_ dbConn rcvQueue
+    insertRcvConnection_ dbConn rcvQueue
+
+createSndQueueAndConn :: DB.Connection -> SendQueue -> IO ()
+createSndQueueAndConn dbConn sndQueue =
+  DB.withTransaction dbConn $ do
+    upsertServer_ dbConn (server (sndQueue :: SendQueue))
+    insertSndQueue_ dbConn sndQueue
+    insertSndConnection_ dbConn sndQueue
+
+upsertServer_ :: DB.Connection -> SMPServer -> IO ()
+upsertServer_ dbConn SMPServer {host, port, keyHash} = do
+  let port_ = serializePort_ port
+  DB.executeNamed
+    dbConn
+    upsertServerQuery_
+    [":host" := host, ":port" := port_, ":key_hash" := keyHash]
+
+upsertServerQuery_ :: Query
+upsertServerQuery_ =
+  [sql|
+    INSERT INTO servers (host, port, key_hash) VALUES (:host,:port,:key_hash)
+    ON CONFLICT (host, port) DO UPDATE SET
+      host=excluded.host,
+      port=excluded.port,
+      key_hash=excluded.key_hash;
+  |]
 
-getServer :: (MonadUnliftIO m, MonadError StoreError m) => SQLiteStore -> SMPServerId -> m SMPServer
-getServer SQLiteStore {conn} serverId = do
-  r <-
-    liftIO $
-      DB.queryNamed
-        conn
-        "SELECT host, port, key_hash FROM servers WHERE server_id = :server_id"
-        [":server_id" := serverId]
-  case r of
-    [smpServer] -> return smpServer
-    _ -> throwError SENotFound
+insertRcvQueue_ :: DB.Connection -> ReceiveQueue -> IO ()
+insertRcvQueue_ dbConn ReceiveQueue {..} = do
+  let port_ = serializePort_ $ port server
+  DB.executeNamed
+    dbConn
+    insertRcvQueueQuery_
+    [":host" := host server, ":port" := port_, ":rcv_id" := rcvId, ":conn_alias" := connAlias, ":rcv_private_key" := rcvPrivateKey, ":snd_id" := sndId, ":snd_key" := sndKey, ":decrypt_key" := decryptKey, ":verify_key" := verifyKey, ":status" := status]
```

Comment:
maybe multiline?

Reply:
I thought we'd better set up max_line_length for ormolu and hlint, apparently for ormolu though you're supposed to program the rule yourself

Reply:
Fixed long lines manually for now